lib/commands/cloud-codesign.ts

```diff
@@ -0,0 +1,71 @@
+import { isInteractive } from "../helpers";
+
+export class CloudCodesignCommand implements ICommand {
+	private readonly parametersValidationText = "The command has only two valid parameters - Apple account id and Apple account password.";
+	// Currently only iOS codesign generation is supported.
+	private readonly platform = this.$devicePlatformsConstants.iOS;
+	private devices: Mobile.IDevice[];
+	public allowedParameters: ICommandParameter[];
+
+	constructor(private $logger: ILogger,
+		private $devicePlatformsConstants: Mobile.IDevicePlatformsConstants,
+		private $devicesService: Mobile.IDevicesService,
+		private $errors: IErrors,
+		private $projectData: IProjectData,
+		private $prompter: IPrompter,
+		private $options: IOptions,
+		private $cloudCodesignService: ICloudCodesignService) {
+		this.$projectData.initializeProjectData();
+	}
+
+	public async execute(args: string[]): Promise<void> {
+		let username = args[0];
+		let password = args[1];
+
+		if (!username) {
+			if (isInteractive) {
+				username = await this.$prompter.getString("Apple ID", { allowEmpty: false });
+			} else {
+				this.$errors.fail(this.parametersValidationText);
+			}
+		}
+
+		if (!password) {
+			if (isInteractive) {
+				password = await this.$prompter.getPassword("Apple ID password");
+			} else {
+				this.$errors.fail(this.parametersValidationText);
+			}
+		}
+
+		this.$logger.info("Generating codesign files in the cloud");
+
+		const attachedDevices: Mobile.ISimpleDevice[] = this.devices.map(d => ({ displayName: d.deviceInfo.displayName, identifier: d.deviceInfo.identifier }));
```

Comment:
can't we make the attachedDevices just instance of `Mobile.IDeviceInfo` and map should be: `this.devices.map (d => d.deviceInfo);`
Also where is this `ISimpleDevice`, I couldn't find it